### PR TITLE
Adds Workflow that triggers Workflow in Campus-Flutter

### PR DESCRIPTION
**Pull request by jakobkoerber**
TODO:
- [x] Add Fine-grained personal access token for Campus-Flutter with `Contents -> Read and Write` and `Pull requests -> Read and Write` permissions to this repository's secrets with the name `CAMPUS_FLUTTER_TOKEN`

### CODE REVIEW

File: .github/workflows/client_update.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Trigger Client Update
+
+on:
+  push:
+    branches:
+      - 'main'
+    paths:
+      - 'server/api/tumdev/**'
```
Comment (CommanderStorm) [inline .github/workflows/client_update.yml:8]: I think limiting this to the file which is actually being used may be smart

```suggestion
      - 'server/api/tumdev/campus_backend.proto'
```

or to limit this to `proto` files
```suggestion
      - 'server/api/tumdev/*.proto'
```

File: .github/workflows/client_update.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Trigger Client Repository Dispatch
+        uses: peter-evans/repository-dispatch@v2
```
Comment (CommanderStorm) [inline .github/workflows/client_update.yml:18]: From a conceptual perspective:
Is there a reason why adding this extra step is better?

I think leaving the complexity of adding the PR in this repo might be a good approach, as this way the boundarys are clearer.

=> i.e. something like https://github.com/orgs/community/discussions/49745

File: .DS_Store
Comment (CommanderStorm) [inline .DS_Store:1]: What is this files job?

File: .github/workflows/client_update.yml (hunk 2 above)
Comment (jakobkoerber) [reply .github/workflows/client_update.yml:18]: What exactly do you mean?

**Pull request merged**